### ISSUE TITLE
VariableMinLength is in the wrong category

### ISSUE THREAD

**Issue by amagdaleno** [arturbosch/detekt#843]
## Expected Behavior
Defining VariableMinLength under naming (as the default configuration does) should work, but doesn't

## Current Behavior
If defined under naming the setting is ignored, instead it uses a default value of 3? and is reported under style instead of naming. Defining VariableMinLength under style does work.

## Steps to Reproduce (for bugs)
Use default configuration and create a variable with an identifier length of 1 or 2

## Context
It should go under naming to be honest

## Your Environment
* Version used: 1.0.0.RC6-4

**Comment by arturbosch** [comment]
Totally right. Thanks for reporting!

**Comment by arturbosch** [comment]
Could not reproduce it.

```
naming:
  VariableMinLength:
      active: true
      minimumVariableNameLength: 3
```
this configuration works for me and reports:

```
Ruleset: naming - 10min debt
	TopLevelPropertyNaming - [a] at /home/artur/Repos/detekt/detekt-rules/src/main/kotlin/io/gitlab/arturbosch/detekt/rules/naming/NamingRules.kt:97:1
	VariableMinLength - [a] at /home/artur/Repos/detekt/detekt-rules/src/main/kotlin/io/gitlab/arturbosch/detekt/rules/naming/NamingRules.kt:97:1
```

Are you sure you used RC6-4? Some versions ago we moved all naming rules from style to the new ruleset ...

**Comment by amagdaleno** [comment]
Truth be told I am using detekt through https://github.com/vanniktech/gradle-code-quality-tools-plugin so if you can't reproduce the issue might stem from there. I've seen the author @vanniktech is active on this project when trying to debug this issue, so maybe he can shed some light on the matter, but if you'd rather go ahead and close this, I can open a ticket there.

And to answer your question I supposedly am, relevant configuration for codeQualityTools looks like this:
```
codeQualityTools {
    detekt {
        version = '1.0.0.RC6-4'
    }
}
```

**Comment by vanniktech** [comment]
```
codeQualityTools {
    detekt {
        version = '1.0.0.RC6-4'
    }
}
```

is indeed correct. Is your yml configuration file for detekt under `code_quality_tools/detekt.yml`?

**Comment by vanniktech** [comment]
Is the project open source and I can take a look at it? Otherwise it's hard for me to debug. Also make sure to be on 0.10.0 of the Gradle Plugin.
Otherwise this works fine for me.

**Comment by amagdaleno** [comment]
Unfortunately the project is not open source. I'm indeed using version 0.10.0:
`classpath 'com.vanniktech:gradle-code-quality-tools-plugin:0.10.0'`

I can set up a small project this afternoon to demonstrate if you want. If it's any help a detekt.yml containing only:
```
autoCorrect: false
failFast: true

naming:
  VariableMinLength:
    active: false
```
fails with a variable length of 2, but this one doesn't:

```
autoCorrect: false
failFast: true

style:
  VariableMinLength:
    active: false
```

**Comment by vanniktech** [comment]
It works for me correctly. Here's a branch where you can play around with it - https://github.com/vanniktech/junit-rules/tree/VariableMinLength

Note though that `VariableMinLength`'s `minimumVariableNameLength` has a default value of 1 which is always true.

**Comment by schalkms** [comment]
Can we close this issue?

**Comment by vanniktech** [comment]
I'd say yes.

**Comment by vanniktech** [comment]
Worst case we're continuing in an issue on my repo.

**Comment by arturbosch** [comment]
Ok, if it turns out to be an issue in detekt feel free to reopen :)

**Comment by vanniktech** [comment]
@arturbosch I've debugged it and the problem is only when you also use `failFast: true`. In case you don't the rule is working as intended.

**Comment by arturbosch** [comment]
Ok, any idea what is causing it? You had a minimal example repository?
